app-engine/demo-version-index/main.py

```diff
@@ -0,0 +1,162 @@
+# Shaka Player Version Index - Appspot Entrypoint
+# Copyright 2022 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+# Generate an index of Shaka Player versions on appspot.
+
+import collections
+import google.appengine.api.modules
+import jinja2
+import os
+import random
+import re
+
+from flask import Flask, render_template
+
+DEMO_URL_TEMPLATE = 'https://{0}-dot-shaka-player-demo.appspot.com/'
+
+# In Google Hosted Libraries
+HOSTED_URL_TEMPLATE = 'https://ajax.googleapis.com/ajax/libs/shaka-player/{0}/shaka-player.{1}'
+
+# Before Google Hosted Libraries, v1 appspot URLs
+V1_URL_TEMPLATE = 'https://{0}-dot-shaka-player-demo.appspot.com/shaka-player.{1}'
+# Before Google Hosted Libraries, v2 and v3 appspot URLs
+V2_URL_TEMPLATE = 'https://{0}-dot-shaka-player-demo.appspot.com/dist/shaka-player.{1}'
+
+def version_to_demo_url(v):
+  return DEMO_URL_TEMPLATE.format(v.replace('.', '-'))
+
+def version_to_lib_url(v):
+  if v == 'nightly':
+    return V2_URL_TEMPLATE.format(v.replace('.', '-'), 'compiled.js')
```

Comment:
If v == 'nightly', do we really need to replace all of the dots with dashes?

Reply:
No, not at all.  Fixed.